tests/test_api_routes.py

```diff
@@ -227,6 +227,121 @@ async def test_simple_chat_flow(test_client):
     assert response_data["session_state"] is None
 
 
+@pytest.mark.asyncio
+async def test_simple_chat_streaming_flow(test_client):
+    """test the simple chat streaming flow route with hybrid retrieval mode"""
+    response = test_client.post(
+        "/chat/stream",
+        json={
+            "context": {
+                "overrides": {"top": 1, "use_advanced_flow": False, "retrieval_mode": "hybrid", "temperature": 0.3}
+            },
+            "messages": [{"content": "What is the capital of France?", "role": "user"}],
+        },
+    )
+    response_data = response.content.split(b"\n")
+    assert response.status_code == 200
+    assert response.headers["Content-Type"] == "application/x-ndjson"
+    assert response_data[0] == (
```

Comment:
Can we use snapshot testing here? Ah I guess we dont use pytest-snapshot in this repo yet, but its quite good for this sort of thing.

Reply:
I will add snapshot, I think it's better than this the file is huge now.